### PR TITLE
[SDK] Add basic Canary strategy integration tests

### PR DESCRIPTION
**Pull request by mbdcos**

### CODE REVIEW

**Comment by mbdcos** [comment]
retest this please

File: frameworks/kafka/tests/test_shakedown_basics.py [hunk 1]
```diff
@@ -83,19 +88,19 @@ def test_broker_invalid():
 @pytest.mark.special
 def test_pods_restart():
     for i in range(DEFAULT_BROKER_COUNT):
-        broker_id = tasks.get_task_ids(SERVICE_NAME,'{}-{}-broker'.format(POD_TYPE, i))
-        restart_info = service_cli('pods restart {}-{}'.format(POD_TYPE, i))
-        tasks.check_tasks_updated(SERVICE_NAME, '{}-{}-broker'.format(POD_TYPE, i), broker_id)
+        broker_id = tasks.get_task_ids(SERVICE_NAME,'{}-{}-{}'.format(DEFAULT_POD_TYPE, i, DEFAULT_TASK_NAME))
+        restart_info = service_cli('pods restart {}-{}'.format(DEFAULT_DEFAULT_POD_TYPE, i))
```
Comment (susanxhuynh) [inline frameworks/kafka/tests/test_shakedown_basics.py]: Typo? "DEFAULT_DEFAULT_POD_TYPE"

File: frameworks/kafka/tests/test_shakedown_strategy.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+def setup_module(module):
+    install.uninstall(SERVICE_NAME, PACKAGE_NAME)
+    utils.gc_frameworks()
+    shakedown.install_package(PACKAGE_NAME,
+                              service_name=SERVICE_NAME,
+                              options_json=install.get_package_options(
+                                  additional_options=DEPLOY_STRATEGY_SERIAL_CANARY
+                              ),
+                              wait_for_completion=True)
```
Comment (susanxhuynh) [inline frameworks/kafka/tests/test_shakedown_strategy.py]: Did you consider using sdk_install::install() here instead? Just wanted to point it out.

File: frameworks/kafka/tests/test_shakedown_strategy.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+    assert pl['phases'][0]['steps'][0]['status'] == 'WAITING'
+    assert pl['phases'][0]['steps'][1]['status'] == 'WAITING'
+    if DEFAULT_BROKER_COUNT > 2:
+        assert pl['phases'][0]['steps'][2]['status'] == 'PENDING'
+
+
+@pytest.mark.smoke
+@pytest.mark.sanity
+def test_canary_fist():
```
Comment (susanxhuynh) [inline frameworks/kafka/tests/test_shakedown_strategy.py]: Typo? s/"first"/"fist"

File: frameworks/kafka/tests/test_shakedown_strategy.py [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+        assert False, "Should not start task now"
+    except AssertionError as arg:
+        raise arg
+    except:
+        pass  # expected to fail
+
+    tasks.check_running(SERVICE_NAME, DEFAULT_BROKER_COUNT)
+
+    try:
+        tasks.check_running(SERVICE_NAME, DEFAULT_BROKER_COUNT + 1, timeout_seconds=60)
```
Comment (susanxhuynh) [inline frameworks/kafka/tests/test_shakedown_strategy.py]: Is this check necessary? It looks like the same check was done in Line 149?

File: frameworks/kafka/tests/test_shakedown_basics.py (hunk 1 above)
Comment (mbdcos) [inline frameworks/kafka/tests/test_shakedown_basics.py]: Yes, this is a bug. Thank you

File: frameworks/kafka/tests/test_shakedown_strategy.py (hunk 2 above)
Comment (mbdcos) [inline frameworks/kafka/tests/test_shakedown_strategy.py]: For canary, no task will start at first.  May be can pass running_task_count as 0. Still need to modify install to skip the rest if running_task_count is 0

File: frameworks/kafka/tests/test_shakedown_strategy.py (hunk 4 above)
Comment (mbdcos) [inline frameworks/kafka/tests/test_shakedown_strategy.py]: Deleted. Thank you.

**Pull request merged**